### PR TITLE
fix(EventBus): Fix the issue that EventBus does not support multiple Publish

### PR DESCRIPTION
**Pull request by zhenlei520**
# Description

1. Fixed that only the first data will be saved after calling EventBus.PublishAsync multiple times 
2. After repairing the data exception, if there are still transaction submissions in the future, the problem of data rollback error

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Extended the documentation

### CODE REVIEW

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/EventBus.cs [hunk 1]
```diff
@@ -64,4 +68,6 @@ public async Task CommitAsync(CancellationToken cancellationToken = default)
 
         await _unitOfWork.CommitAsync(cancellationToken);
     }
+
+    // public bool IsFirst() =>
```
Comment (doddgu) [inline src/Dispatcher/Masa.Contrib.Dispatcher.Events/EventBus.cs:72]: Remove the comment.

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/Internal/InitializeServiceProvider.cs [hunk 2]
```diff
@@ -0,0 +1,26 @@
+﻿// Copyright (c) MASA Stack All rights reserved.
+// Licensed under the MIT License. See LICENSE.txt in the project root for license information.
+
+namespace Masa.Contrib.Dispatcher.Events.Internal;
+
+internal class InitializeServiceProvider : IInitializeServiceProvider
+{
+    private bool _state;
```
Comment (doddgu) [inline src/Dispatcher/Masa.Contrib.Dispatcher.Events/Internal/InitializeServiceProvider.cs:8]: What's the state mean? readyToRun? initialized?

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/README.zh-CN.md [hunk 3]
```diff
@@ -210,4 +210,8 @@ IEventBus是事件总线的核心，配合Cqrs、Uow、Masa.Contrib.Ddd.Domain.R
 
 > 问题4. 为什么开启了异常重试却未执行重试？
 
-    > 默认`UserFriendlyException`不支持重试，如果需要支持重试，则需要重新实现`IStrategyExceptionProvider`
\ No newline at end of file
+    > 默认`UserFriendlyException`不支持重试，如果需要支持重试，则需要重新实现`IStrategyExceptionProvider`
```
Comment (doddgu) [inline src/Dispatcher/Masa.Contrib.Dispatcher.Events/README.zh-CN.md:213]: IStrategyExceptionProvider?  IExceptionStrategyProvider

**Review by doddgu — CHANGES_REQUESTED** [review]

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/EventBus.cs (hunk 1 above)
Comment (zhenlei520) [reply src/Dispatcher/Masa.Contrib.Dispatcher.Events/EventBus.cs]: Processed

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/Internal/InitializeServiceProvider.cs (hunk 2 above)
Comment (zhenlei520) [reply src/Dispatcher/Masa.Contrib.Dispatcher.Events/Internal/InitializeServiceProvider.cs]: Processed

File: src/Dispatcher/Masa.Contrib.Dispatcher.Events/README.zh-CN.md (hunk 3 above)
Comment (zhenlei520) [reply src/Dispatcher/Masa.Contrib.Dispatcher.Events/README.zh-CN.md]: Processed

**Review by doddgu — APPROVED** [review]
LGTM

**Pull request merged**